File: Step 1 - Learn the Basics/1.2 Build-up Logical Thinking/01 Patterns/ForestVisualizer.java

import java.util.Scanner;

public class ForestVisualizer {
    public static void main(String[] args) {
        // Create scanner object to get input
        Scanner scanner = new Scanner(System.in);

        // Input N
        int N = scanner.nextInt();
        
        // Loop to generate forest pattern
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
        
        // Close the scanner
        scanner.close();
    }
}

File: Step 1 - Learn the Basics/1.2 Build-up Logical Thinking/01 Patterns/forest_visualizer.py
# https://takeuforward.org/strivers-a2z-dsa-course/must-do-pattern-problems-before-starting-dsa/
# Python Script to generate N-dimensional forest pattern
def print_forest(N):
    for _ in range(N):
        print('* ' * N)

# Get the input
N = int(input())

# Print the forest pattern
print_forest(N)
